CMake/RealmCore.cmake

```diff
@@ -179,7 +180,11 @@ function(download_realm_core core_version)
         download_realm_tarball(${url_release} ${core_directory_release} ${core_library_release})
     else()
         if(APPLE)
-            set(basename "realm-core-cocoa")
+            if(REALM_CORE_PACKAGING STREQUAL "1")
```

Comment:
My idea was to use a separate flag to make it easier to support a future 2.x release using the updated packaging, but maybe that isn't actually useful (since once such a release is made we'd no longer need to support the old packaging).